api/src/main/java/org/openmrs/module/clientregistry/api/providers/r4/CRPatientFhirResourceProvider.java

```diff
@@ -0,0 +1,81 @@
+package org.openmrs.module.clientregistry.api.providers.r4;
+
+import ca.uhn.fhir.model.api.Include;
+import ca.uhn.fhir.model.valueset.BundleTypeEnum;
+import ca.uhn.fhir.rest.annotation.*;
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.*;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.AllergyIntolerance;
+import org.hl7.fhir.r4.model.DiagnosticReport;
+import org.hl7.fhir.r4.model.Encounter;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.openmrs.module.clientregistry.api.ClientRegistryManager;
+import org.openmrs.module.clientregistry.api.model.CRPatient;
+import org.openmrs.module.fhir2.api.annotations.R4Provider;
+import org.openmrs.module.fhir2.api.search.param.PatientSearchParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.List;
+
+@Component("crPatientFhirR4ResourceProvider")
+@R4Provider
+@Setter(AccessLevel.PACKAGE)
+public class CRPatientFhirResourceProvider implements IResourceProvider {
+	
+	@Autowired
+	private ClientRegistryManager clientRegistryManager;
+	
+	@Override
+	public Class<? extends IBaseResource> getResourceType() {
+		return CRPatient.class;
```

Comment:
maybe the resource here is just Patient and we have `/ws/fhir2/R4/Patient/$ihe-pix?sourceIdentifier=<national-id>` as the endpoint url